### PR TITLE
[instancer] Recalc avgcharwidth after instancing

### PR DESCRIPTION
**Pull request by qxliu76**
Port from https://github.com/fonttools/fonttools/commit/3573cd906bafa46069b64c68e97ea69e62d1c03b

### CODE REVIEW

File: src/hb-ot-os2-table.hh [hunk 1]
```diff
@@ -209,6 +209,26 @@ struct OS2
...
+      unsigned width = _.first;
+      if (width)
+      {
+        total_width += width;
+        num++;
+      }
+    }
+
+    unsigned avg_width = 0;
+    if (num) avg_width = (unsigned) roundf (total_width / num);
```
Comment (behdad) [inline src/hb-ot-os2-table.hh:227]: This looks a bit weird. You might as well write:
```
unsigned avg_width = num ? ... : 0;
```

**Comment by behdad** [comment]
@qxliu76 Any idea why instancer-solver is failing in the distcheck bot? Does it require any fonts that are not added to `Makefile.am`?

**Comment by qxliu76** [comment]
> @qxliu76 Any idea why instancer-solver is failing in the distcheck bot? Does it require any fonts that are not added to `Makefile.am`?

uh...I think I need to add approx() on some expected output float values. I'll fix it.
Weird it's not caught in meson test, I just found some tests are not enabled as well: e.g test-priority-queue, test-multimap etc. Any idea why?

**Comment by behdad** [comment]
> > @qxliu76 Any idea why instancer-solver is failing in the distcheck bot? Does it require any fonts that are not added to `Makefile.am`?
> 
> uh...I think I need to add approx() on some expected output float values. I'll fix it. Weird it's not caught in meson test,

Yeah looks like most tests are not included in meson `compiled_tests`?!


> I just found some tests are not enabled as well: e.g test-priority-queue, test-multimap etc. Any idea why?

No idea. Can you add them please?

**Comment by qxliu76** [comment]
> > > @qxliu76 Any idea why instancer-solver is failing in the distcheck bot? Does it require any fonts that are not added to `Makefile.am`?
> > 
> > 
> > uh...I think I need to add approx() on some expected output float values. I'll fix it. Weird it's not caught in meson test,
> 
> Yeah looks like most tests are not included in meson `compiled_tests`?!
> 
> > I just found some tests are not enabled as well: e.g test-priority-queue, test-multimap etc. Any idea why?
> 
> No idea. Can you add them please?

Fixed solver test. And following Makefile.sources, I moved these tests from noinst_programs to compiled_tests.

**Pull request merged**